src/network/generator/network-generator.c

```diff
@@ -959,60 +971,76 @@ static int parse_cmdline_rd_peerdns(Context *context, const char *key, const cha
         return network_set_dhcp_use_dns(context, "", r);
 }
 
+static int extract_vlan_id(const char *vlan_name, uint16_t *ret) {
+        assert(!isempty(vlan_name));
+        assert(ret);
+
+        /* From dracut.cmdline(7):
+         * We support the four styles of vlan names:
+         *   VLAN_PLUS_VID (vlan0005),
+         *   VLAN_PLUS_VID_NO_PAD (vlan5),
+         *   DEV_PLUS_VID (eth0.0005), and
+         *   DEV_PLUS_VID_NO_PAD (eth0.5). */
+
+        for (const char *p = vlan_name + strlen(vlan_name) - 1; p > vlan_name; p--)
+                if (!ascii_isdigit(*p))
+                        return parse_vlanid(p+1, ret);
+
+        return -EINVAL;
+}
+
 static int parse_cmdline_vlan(Context *context, const char *key, const char *value) {
-        const char *name, *p;
+        _cleanup_free_ char *name = NULL;
+        const char *p;
         NetDev *netdev;
         int r;
 
         assert(context);
         assert(key);
 
+        /* vlan=<vlanname>:<phydevice> */
+
         if (proc_cmdline_value_missing(key, value))
                 return log_debug_errno(SYNTHETIC_ERRNO(EINVAL), "Missing value for '%s'", key);
```

Comment:
Not your fault. But `proc_cmdline_value_missing` logs at warning level on its own. All uses in this file seems inappropriate?

Reply:
I.e. either remove the duplicated debug log or step away from `proc_cmdline_value_missing`?